mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1001,11 +996,19 @@ private[spark] object RandomForest extends Logging {
     } else {
       val numSplits = metadata.numSplits(featureIndex)
 
-      // get count for each distinct value
-      val (valueCountMap, numSamples) = featureSamples.foldLeft((Map.empty[Double, Int], 0)) {
+      // get count for each distinct value except zero value
+      val (partValueCountMap, partNumSamples) = featureSamples.foldLeft((Map.empty[Double, Int], 0)) {
         case ((m, cnt), x) =>
           (m + ((x, m.getOrElse(x, 0) + 1)), cnt + 1)
       }
+
+      // Calculate the number of samples for finding splits
+      val requiredSamples: Long = (samplesFractionForFindSplits(metadata) * metadata.numExamples.toDouble).toLong
```

Comment:
I think here it could be:
```
val numSamples: Int = (samplesFractionForFindSplits(metadata) * metadata.numExamples).toInt
```

Reply:
Thanks for your comments, and im sorry for my carelessness. I have fixed them all.